### ISSUE TITLE
http.rb: RuntimeError: Cannot convert a primitive value (type: class java.lang.Boolean, value: false) to an LLVMNativePointer)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1549]
To reproduce:

```ruby
require 'http'

HTTP.get('https://github.com')
```

Error:

```
RuntimeError: Cannot convert a primitive value (type: class java.lang.Boolean, value: false) to an LLVMNativePointer). (LLVMPolyglotException)
Translated to internal error
	from ~/.rubies/truffleruby-1.0.0-rc11/lib/truffle/truffle/cext.rb:221:in `execute_without_conversion'
	from ~/.rubies/truffleruby-1.0.0-rc11/lib/truffle/truffle/cext.rb:221:in `execute_with_mutex'
	from ~/.rubies/truffleruby-1.0.0-rc11/lib/truffle/truffle/cext.rb:1197:in `__allocate__'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/response/parser.rb:9:in `new'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/response/parser.rb:9:in `initialize'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/connection.rb:40:in `new'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/connection.rb:40:in `initialize'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/client.rb:71:in `new'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/client.rb:71:in `perform'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/client.rb:31:in `request'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/chainable.rb:75:in `request'
	from ~/.gem/truffleruby/2.4.4/gems/http-4.0.3/lib/http/chainable.rb:20:in `get'
	from (irb):2
	from ~/.rubies/truffleruby-1.0.0-rc11/bin/irb:29:in `<main>'
```

**Comment by nirvdrum** [comment]
Thanks for the bug report. This issue looks to be resolved with the forthcoming TruffleRuby 1.0.0 RC 12 release. We've substantially improved our handling of native extensions in general. I am unable to reproduce the reported issue using TruffleRuby master.

**Comment by nirvdrum** [comment]
I'm going to mark this as fixed. I'll follow-up on the issue to let you know when the fix is publicly available. Thanks again!
